src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -102,6 +102,23 @@ message Metrics {
     map<string, uint64> counters = 2;
 }
 
+enum FailureScope {
+    FAILURE_SCOPE_UNKNOWN = 0;
+    FAILURE_SCOPE_TASK = 1;        // The task failed, but can be retried.
+    FAILURE_SCOPE_INVOCATION = 2;  // The invocation cannot continue (e.g. invalid input).
+    FAILURE_SCOPE_GRAPH = 3;       // The graph is broken, requiring repair; block invocations.
+}
+
+message FailureInfo {
+    // Scope determines the overall system's behavior in response to this failure.
+    optional FailureScope scope = 1;
+
+    // Cls, Msg, and Trace are human-comprehensible diagnostics.
+    optional string cls = 2;
+    optional string msg = 3;
```

Comment:
All the human comprehensive diagnostics is currently going into stderr.
With this change we're going very inconsistent with how we report human readable data. I think we should delete it from here and just add it to stderr until we decided that we're not using stderr and I won't expect this to happen for now.